Sources/Datadog/Core/Persistence/Files/Directory.swift

```diff
@@ -28,10 +28,16 @@ internal struct Directory {
         return File(url: fileURL)
     }
 
-    /// Returns file with given name.
+    /// Checks if a file with given `fileName` exists in this directory.
+    func hasFile(named fileName: String) -> Bool {
+        let fileURL = url.appendingPathComponent(fileName, isDirectory: false)
+        return FileManager.default.fileExists(atPath: fileURL.path)
+    }
+
+    /// Returns file with given name or throws an error if file does not exist.
     func file(named fileName: String) throws -> File {
         let fileURL = url.appendingPathComponent(fileName, isDirectory: false)
-        guard FileManager.default.fileExists(atPath: fileURL.path) else {
+        guard hasFile(named: fileName) else {
             throw InternalError(description: "File does not exist at path: \(fileURL.path)")
```

Comment:
maybe this is not an error state.? does it make sense if this method returns an optional in case that file doesn't exist?
i'll leave another comment at calling site below

Reply:
This is a core logic in our data persistence and IMO the strickier we are, the better. We know exactly when the file is supposed to exist and when it should / could be gone. By throwing the error here, we can control data consistency more and base less on assumptions. How does it sound?